lib/src/domain/usecase/fetch_genres_usecase.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:http/http.dart';
+
+import '../../core/usecase/i_usecase.dart';
+import '../../core/util/data_state.dart';
+import '../../core/util/enums.dart';
+import '../../data/repository/api_repository.dart';
+import '../entity/genre.dart';
+import '../repository/i_my_repository.dart';
+
+class FetchGenresUseCase
+    extends IUseCase<Future<DataState<List<Genre>>>, List<int>> {
+  final IMyRepository repository;
+
+  FetchGenresUseCase({IMyRepository? repository})
+      : repository = repository ?? APIRepository(client: Client());
+
+  @override
+  Future<DataState<List<Genre>>> call([List<int>? params]) async {
+    DataState<List<Genre>> genresJson = await repository.loadGenres();
+    DataState<List<Genre>> genreNames =
+        DataFailure<List<Genre>>(error: Exception(''));
+    if (genresJson.state == Status.success && params != null) {
```

Comment:
This is something that you don't need to manage here, DataFailure should be received from data layer

Reply:
The UseCase don't manage data changes, the only logic that you are going to have here is in the case that you manage data from api and local database

Reply:
So, what you mean is i have to move the logic to de API repository? Anyway in the BLoC i have and if because i have to check if the parameters are null or not.

Reply:
Basically the UseCase should not have this kind of logic, the only thing that you need to check here is to call the repository